lib/sisimai/bite/json/amazonses.rb

```diff
@@ -0,0 +1,255 @@
+      BounceType = {
+        :Permanent => {
+          :General    => '',
+          :NoEmail    => '',
+          :Suppressed => '',
+        },
+        :Transient => {
+          :General            => '',
+          :MailboxFull        => 'mailboxfull',
+          :MessageTooLarge    => 'mesgtoobig',
+          :ContentRejected    => '',
+          :AttachmentRejected => '',
+        },
+      }
+
+      def smtpagent;   return Sisimai::Bite.smtpagent(self); end
+      def description; return 'Amazon SES(JSON): http://aws.amazon.com/ses/'; end
+
+      # Parse bounce messages from Amazon SES(JSON)
+      # @param         [Hash] mhead       Message headers of a bounce email
+      # @options mhead [String] from      From header
+      # @options mhead [String] date      Date header
+      # @options mhead [String] subject   Subject header
+      # @options mhead [Array]  received  Received headers
+      # @options mhead [String] others    Other required headers
+      # @param         [String] mbody     Message body of a bounce email
+      # @return        [Hash, Nil]        Bounce data list and message/rfc822
+      #                                   part or nil if it failed to parse or
+      #                                   the arguments are missing
+      # @since v4.20.0
+      def scan(mhead, mbody)
+        return nil unless mhead
+        return nil unless mbody
+        return nil unless mhead['x-amz-sns-message-id']
+        return nil unless mhead['x-amz-sns-message-id'].size > 0
+
+        hasdivided = mbody.split("\n")
+        jsonstring = ''
+        jsonthings = nil
+        foldedline = false
+
+        hasdivided.each do |e|
+          # Find JSON string from the message body
+          next if e.size.zero?
+          break if e =~ /\A[-]{2}\z/
+          break if e == '__END_OF_EMAIL_MESSAGE__'
+
+          # The line starts with " ", continued from !\n.
+          e = e.sub(/\A[ ]/, '') if foldedline
+          foldedline = false
+
+          if e =~ /[!]\z/
+            # ... long long line ...![\n]
+            e = e.sub(/!\z/, '')
+            foldedline = true
+          end
+          jsonstring += e
+        end
+
+        begin
+          if RUBY_PLATFORM =~ /java/
+            # java-based ruby environment like JRuby.
+            require 'jrjackson'
+            jsonobject = JrJackson::Json.load(jsonstring)
+            if jsonobject['Message']
+              # 'Message' => '{"notificationType":"Bounce",...
+              jsonthings = JrJackson::Json.load(jsonobject['Message'])
+            end
+          else
+            # Matz' Ruby Implementation
+            require 'oj'
+            jsonobject = Oj.load(jsonstring)
+            if jsonobject['Message']
+              # 'Message' => '{"notificationType":"Bounce",...
+              jsonthings = Oj.load(jsonobject['Message'])
+            end
+          end
+          jsonthings ||= jsonobject
+
```

Comment:
__[rubocop]__
* Layout/EmptyLinesAroundExceptionHandlingKeywords: Extra empty line detected before the `rescue`. [:ref](https://github.com/bbatsov/ruby-style-guide#empty-lines-around-bodies)

<!-- Comment by [SideCI](https://sideci.com) -->